erts/emulator/beam/erl_term_hashing.c

```diff
@@ -1977,38 +2186,50 @@ Uint32 erts_dbg_hashmap_collision_bonanza(Uint32 hash, Eterm key)
         };
         hash = hashvec[hash % (sizeof(hashvec) / sizeof(hashvec[0]))];
     }*/
-    const Uint32 bad_hash = (hash & 0x12482481) * 1442968193;
-    const Uint32 bad_bits = hash % 67;
-    if (bad_bits < 32) {
+    const erts_ihash_t bad_hash =
+        (hash & (erts_ihash_t)0x1248248112482481ull) *
+        (erts_ihash_t)0x5601F2815601F281ull;
```

Comment:
You have increased number of bad hashes to 65536.
I think 256 bad hashes was a good number. It will exercise collision nodes in a lot more test cases that do not create massive maps.